packages/preferences/src/browser/views/preference-scope-tabbar-widget.tsx

```diff
@@ -66,8 +66,15 @@ export class PreferencesScopeTabBar extends TabBar<Widget> implements StatefulWi
         return this.currentSelection;
     }
 
-    protected setNewScopeSelection(newSelection: Preference.SelectedScopeDetails): void {
+    get availableTitles(): Title.Dataset[] {
```

Comment:
Rather than exposing the available titles here, I think I'd prefer a public method that can accept a scope and then do the title-seeking logic locally. Indeed, I prefer it so much I've implemented it in [another PR](https://github.com/eclipse-theia/theia/pull/9439/files) 😉:

![image](https://user-images.githubusercontent.com/62660806/117215092-5cacf980-adc3-11eb-96b3-87bbe60e8788.png)

That code assumes some changes of type - on current master you would just have to look for the title and then pass its dataset into the existing `setNewScopeSelection` method. For full generality, it might be nice to be able to pass a folder URI and then get into folder scope, but neither your case nor mine requires that.